src/common.ts

```diff
@@ -7,7 +7,7 @@ import { AddressProvider } from "./services/addressProvider";
 import { Address } from "./types";
 import { Yearn } from "./yearn";
 
-export class Service<E = {}> {
+export class Service<E = Record<string, unknown>> {
```

Comment:
`{}` stands for "any non nullish value" so the `Record<string, unknown>` is the way to go